### DISCUSSION TITLE
Have to log in with 2FA at least once per day

### DISCUSSION THREAD

**Discussion by jonsterling** [community/community#52958] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

I am very frustrated with the forced transition to 2FA; one of the problems I am having is that I have to log in every single day at least once, since (for some reason) GitHub keeps logging me out. I would like to ask that the sessions time out on a much longer interval, preferably at least a week. It is not acceptable that every single morning when I get online to start coding, I have to go and find my phone in enough time and open the stupid GitHub app to log in. This is a severe UX regression, and it is leading me to move all my work off of GitHub. Is anyone else having this problem?

**Comment by airtower-luna** [comment]
That's odd, sessions do last a lot longer in my experience. As you mention "every morning", is there any chance that

* you shut down your computer (or otherwise close the browser) in the evening and
* your browser is configured to clear cookies when closed (by browser settings or an add-on)?

Because that combination would lead to the login session getting lost.

If that's the case, you could change the settings to keep cookies from `github.com` (and possibly other sites where you want to stay logged in). Mind there's a security tradeoff, if someone could get access to your browser data with the stored cookies they could act as you. On a personal system with disk encryption that's probably fine, on a shared system maybe not so much.

**Reply by jonsterling** [reply]
Hi, thanks & I appreciate you chiming in! Unfortunately I do not have any setting to clear cookies. I am guessing this must be some unintentional effect that could be hard to track down, but hopefully if anyone else is having this problem, we could help diagnose what's going on.

Security is great, but my experience with things like 2FA is that it is making me want to wear a t-shirt that has my recovery codes written on it in big letters.
